src/ragQueries.ts

```diff
@@ -0,0 +1,42 @@
+import { GraphAnnotation } from "./agent/agent"
+import { OpenAIEmbeddings } from "@langchain/openai";
+import { Chroma } from "@langchain/community/vectorstores/chroma";
+
+const embeddings = new OpenAIEmbeddings({
+    model: "text-embedding-3-small",
+    apiKey: process.env.OPENAI_API_KEY || ""
+});
+
+const vectorStore = new Chroma(embeddings, {
+    collectionName: "uw-courses-test",
+    url: "http://localhost:8080", // Optional, will default to this value
+    collectionMetadata: {
+      "hnsw:space": "cosine",
+    }, // Optional, can be used to specify the distance method of the embedding space https://docs.trychroma.com/usage-guide#changing-the-distance-function
+  });
+
+
+// Thoughts: can probably turn this into a retreiver: https://js.langchain.com/docs/integrations/vectorstores/chroma/#query-by-turning-into-retriever
+// vector similarity search 
+export const vectorSimilaritySearch = async (state: typeof GraphAnnotation.State) => {
+    console.log("vectorSimilaritySearch");
+
+    // perform vector similarity search
+    // vectorStore.similaritySearchWithScore can be used to get scores if we want to filter by score
```

Comment:
@jonhuo11 should I use similaritySearchWithScore and filter based on sum score? I think for now lets not do it, but interesting that we get this capability

Reply:
We should search for similar results, take the top 5-10 results, and then pass them as reference material into the prompt for answering the user question. So id say we probably need to rank by score and in our prompt explain how the order of the documents is based on relevance score

Reply:
Also we should probably sync with Arnav/Jerome and figure out what format they have been writing the documents into the DB, and also when we pass the retrieved documents into the prompt we need to also have some format:


```
(example of a prompt here)
Using the relevant courses and their information below, try to answer the users question:

User question:
<User question>


(we would inject these retrieved courses into the prompt here)
------------
COURSE: <Course name>
DESCRIPTION: <Description>
OFFERED: <Term of next offering>
PREREQS: <Prereqs>
ANTIREQS: <Antireqs>
------------
COURSE: <Course name>
DESCRIPTION: <Description>
OFFERED: <Term of next offering>
PREREQS: <Prereqs>
ANTIREQS: <Antireqs>
------------
COURSE: <Course name>
DESCRIPTION: <Description>
OFFERED: <Term of next offering>
PREREQS: <Prereqs>
ANTIREQS: <Antireqs>
------------
....
```